homeassistant/components/octoprint/config_flow.py

```diff
@@ -0,0 +1,203 @@
+"""Config flow for OctoPrint integration."""
+import logging
+from urllib.parse import urlsplit
+
+from pyoctoprintapi import ApiError, OctoprintClient
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_HOST,
+    CONF_PATH,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _schema_with_defaults(username="", host=None, port=80, path="/", ssl=False):
+    return vol.Schema(
+        {
+            vol.Required(CONF_USERNAME, default=username): cv.string,
+            vol.Required(CONF_HOST, default=host): cv.string,
+            vol.Optional(CONF_PORT, default=port): cv.port,
+            vol.Optional(CONF_PATH, default=path): cv.string,
+            vol.Optional(CONF_SSL, default=ssl): cv.boolean,
+        },
+        extra=vol.ALLOW_EXTRA,
+    )
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+
+    session = async_get_clientsession(hass)
+    octoprint = OctoprintClient(
+        data[CONF_HOST], session, data[CONF_PORT], data[CONF_SSL], data[CONF_PATH]
+    )
+    octoprint.set_api_key(data[CONF_API_KEY])
+
+    uuid = None
+
+    try:
+        discovery = await octoprint.get_discovery_info()
+        if discovery:
```

Comment:
Please move lines that can't raise out of the try... except block.

Reply:
Done